### PR TITLE
Word256 support

### PR DESCRIPTION
**Pull request by Zankoku-Okuno**

### CODE REVIEW

File: src/Basics/Word256.hs [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+index# :: ByteArray# -> Int# -> T#
+index# arr# i# =
+  (# Exts.indexWordArray# arr# (4# *# i#)
+  ,  Exts.indexWordArray# arr# ((4# *# i#) +# 1#)
+  ,  Exts.indexWordArray# arr# ((4# *# i#) +# 2#)
+  ,  Exts.indexWordArray# arr# ((4# *# i#) +# 3#)
+  #)
+
+read# :: MutableByteArray# s -> Int# -> State# s -> (# State# s, T# #)
+read# arr# i# s0 = case Exts.readWordArray# arr# (2# *# i#) s0 of
```
Comment (andrewthad) [inline src/Basics/Word256.hs]: Multiply by four here, not by two.

File: src/Basics/Word256.hs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  (# s1, i0 #) -> case Exts.readWordArray# arr# ((2# *# i#) +# 1#) s1 of
+    (# s2, i1 #) -> case Exts.readWordArray# arr# ((2# *# i#) +# 2#) s2 of
+      (# s3, i2 #) -> case Exts.readWordArray# arr# ((2# *# i#) +# 3#) s3 of
+        (# s4, i3 #) -> (# s4, (# i0, i1, i2, i3 #) #)
+
+write# :: MutableByteArray# s -> Int# -> T# -> State# s -> State# s
+write# arr# i# (# a, b, c, d #) s0 =
+  case Exts.writeWordArray# arr# (2# *# i#) a s0 of
```
Comment (andrewthad) [inline src/Basics/Word256.hs]: Same issue, multiply by four, not by two.

**Review by andrewthad — CHANGES_REQUESTED** [review]
Mostly good. Just two small issues.

File: src/Basics/Word256.hs (hunk 2 above)
Comment (Zankoku-Okuno) [reply src/Basics/Word256.hs]: See? #2

Comment (andrewthad) [reply src/Basics/Word256.hs]: Agreed. That would help catch these kinds of problems.

**Pull request merged**